vexim/sitepasswordsubmit.php

```diff
@@ -11,13 +11,18 @@
     $success = $sth->execute(array(':crypt'=>$cryptedpassword));
     if ($success) {
       $_SESSION['crypt'] = $cryptedpassword;
+      if(isset($_SESSION['pw_enforce'])) unset($_SESSION['pw_enforce']);
       header ("Location: site.php?sitepass=success");
       die;
     } else {
       header ("Location: site.php?sitepass=fail");
       die;
     }
   } else {
+    if(isset($_SESSION['pw_enforce'])) {
+      header ("Location: sitepassword.php?badpass=siteadmin");
+      die;
+    }
     header ("Location: site.php?badpass=siteadmin");
```

Comment:
Yes, could do it. However, if you changed your mind about changing the 
password (or it doesn't fit the security requirements of the 
password-policy), you are somehow trapped. You could just logout.

On 2016-08-03 12:04, Rimas Kudelis wrote:
> In vexim/sitepasswordsubmit.php [1]:
> 
>>        header ("Location: site.php?sitepass=success");
>>        die;
>>      } else {
>>        header ("Location: site.php?sitepass=fail");
>>        die;
>>      }
>>    } else {
>> +    if(isset($_SESSION['pw_enforce'])) {
>> +      header ("Location: sitepassword.php?badpass=siteadmin");
>> +      die;
>> +    }
>>      header ("Location: site.php?badpass=siteadmin");
> 
> Perhaps we should just update this line instead?
> 
> --
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub [2], or mute the
> thread [3].
> 
> 	*
> 
> Links:
> ------
> [1] https://github.com/vexim/vexim2/pull/178#discussion_r73310971
> [2]
> https://github.com/vexim/vexim2/pull/178/files/9bf36be48b51b9cc0030d9e5307da598a940ec5c#r73310971
> [3]
> https://github.com/notifications/unsubscribe-auth/AGCFFHJdG9EiBEs2mmKoHEL3vXhV-ek0ks5qcGe9gaJpZM4JbIdI